chia/util/wallet_tools.py

```diff
@@ -157,20 +161,31 @@ def generate_unsigned_transaction(
             change_puzzle_hash = self.get_new_puzzlehash()
             change_output = ConditionWithArgs(ConditionOpcode.CREATE_COIN, [change_puzzle_hash, int_to_bytes(change)])
             condition_dic[output.opcode].append(change_output)
-            main_solution: Program = self.make_solution(condition_dic)
-        else:
-            main_solution = self.make_solution(condition_dic)
 
+        secondary_coins_cond_dic: Dict[ConditionOpcode, List[ConditionWithArgs]] = dict()
+        secondary_coins_cond_dic[ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT] = []
         for n, coin in enumerate(coins):
             puzzle_hash = coin.puzzle_hash
             if secret_key is None:
                 secret_key = self.get_private_key_for_puzzle_hash(puzzle_hash)
             pubkey = secret_key.get_g1()
             puzzle = puzzle_for_pk(bytes(pubkey))
             if n == 0:
+                message_list = [c.name() for c in coins]
+                for outputs in condition_dic[ConditionOpcode.CREATE_COIN]:
+                    message_list.append(Coin(coin.name(), outputs.vars[0], int_from_bytes(outputs.vars[1])).name())
+                message = std_hash(b"".join(message_list))
+                condition_dic[ConditionOpcode.CREATE_COIN_ANNOUNCEMENT].append(
+                    ConditionWithArgs(ConditionOpcode.CREATE_COIN_ANNOUNCEMENT, [message])
+                )
+                primary_announcement_hash = Announcement(coin.name(), message).name()
+                secondary_coins_cond_dic[ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT].append(
+                    ConditionWithArgs(ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT, [primary_announcement_hash])
+                )
+                main_solution = self.make_solution(condition_dic)
```

Comment:
It's not clear what the argument types are for `make_solution`. We should use python types for all function arguments and return types.
Same for the functions like `make_create_coin_announcement`.
If we don't use types it's very easy to pass in the wrong things

Reply:
That's probably a separate PR though.